sw/device/tests/crypto/aes_gcm_testutils.c

```diff
@@ -0,0 +1,104 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/tests/crypto/aes_gcm_testutils.h"
+
+#include "sw/device/lib/base/macros.h"
+#include "sw/device/lib/crypto/drivers/aes.h"
+#include "sw/device/lib/crypto/impl/aes_gcm/aes_gcm.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+
+uint64_t t_start;
+uint64_t t_end;
+/**
+ * Start a cycle-count timing profile.
+ */
+static void profile_start() { t_start = ibex_mcycle_read(); }
+
+/**
+ * End a cycle-count timing profile.
+ *
+ * Call `profile_start()` first.
+ */
+static void profile_end(const char *profile_name) {
+  t_end = ibex_mcycle_read();
+  uint32_t cycles = t_end - t_start;
```

Comment:
On this line, we do an implicit narrowing conversion from `uint64_t` to `uint32_t`. I think it would make sense to CHECK that the result will fit and do an explicit `(uint32_t)` cast. I already did something similar in the `memory_perftest`: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/base/memory_perftest.c;l=232-234;drc=e00f988e3c6d64697e72d3cb7518f137bb61c7e4